### PR TITLE
Migration to official client-go lib for Kubernetes related agent features.

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

Migrate the kubernetes_apiserver check and underlying functionalities to the official client-go from kubernetes.

### Motivation

Only have 1 client.

### Migration status

- [x] Leader Election
- [x] Component Status Service Check
- [x] Event Collection
- [x] Event Processing
- [x] Unify clients 
- [x] Remove traces of previous library
- [x] Tests up to date
- [x] Stability (let it run for several days without unexpected issues)

### CODE REVIEW

File: pkg/util/kubernetes/apiserver/apiserver.go [hunk 1]
```diff
@@ -78,39 +78,52 @@ func GetAPIClient() (*APIClient, error) {
 	return globalApiClient, nil
 }
 
+// GetClient returns an official Kubernetes core v1 client
+func GetClient() (*corev1.CoreV1Client, error) {
```
Comment (CharlyF) [inline pkg/util/kubernetes/apiserver/apiserver.go]: TODO: Use `GetCoreV1Client` in the util/kubernetes pkg

File: pkg/util/kubernetes/apiserver/events.go [hunk 2]
```diff
@@ -31,82 +28,70 @@ var eventReadTimeout = 100 * time.Millisecond
...
-				*latestResVersion = "0"
+		select {
+		case rcvdEv := <-eventWatcher.ResultChan():
+			if rcvdEv.Type == watch.Error {
+				errEvent, ok := rcvdEv.Object.(*metav1.Status)
+				if !ok {
+					return nil,nil,"0",errors.New("could not parse status of error event from the API Server.") // TODO custom error
+				}
+				if errEvent.Reason == "Expired"{
+					// Known issue with ETCD https://github.com/kubernetes/kubernetes/issues/45506
```
Comment (CharlyF) [inline pkg/util/kubernetes/apiserver/events.go]: Discuss with the team about this one.

**Review by xvello — DISMISSED** [review]
LGTM, two nits

File: pkg/util/kubernetes/apiserver/apiserver.go [hunk 3]
```diff
@@ -77,48 +77,56 @@ func GetAPIClient() (*APIClient, error) {
 	return globalAPIClient, nil
 }
 
+// GetClient returns an official Kubernetes core v1 client
+func GetClient() (*corev1.CoreV1Client, error) {
```
[the remark this answers is not in the document]
Comment (CharlyF) [reply pkg/util/kubernetes/apiserver/apiserver.go]: This is for the leader election and the tests mostly, I agree that this method should be private for connect.
I would need to make a method to get/list configmaps, create/update events (and maybe others I missed).

Do you think it's worth it @xvello  ?

File: pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go [hunk 4]
```diff
@@ -118,7 +118,9 @@ func (le *LeaderEngine) init() error {
 	}
 	log.Debugf("LeaderLeaseDuration: %s", le.LeaseDuration.String())
 
-	le.coreClient, err = apiserver.GetClient()
+	apiClient, err := apiserver.GetAPIClient()
+	le.coreClient = apiClient.Client
```
Comment (xvello) [inline pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go]: You should check the err is not nil before accessing a potentially nil object

**Review by xvello — DISMISSED** [review]
One last change and a nit, then :shipit: :)

**Review by mfpierre — APPROVED** [review]
☸️ 🆕 🆑 💯 ✅

**Pull request merged**